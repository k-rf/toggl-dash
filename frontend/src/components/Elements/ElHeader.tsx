import { AppBar, AppBarProps, Box, Toolbar } from "@mui/material";

type MuiProps = Pick<AppBarProps, "position" | "sx">;
type CustomProps = { title: string; menu?: React.ReactNode };

type Props = MuiProps & CustomProps;

export const ElHeader = (props: Props) => {
  return (
    <AppBar>
      <Toolbar>
        {props.title}
        {props.menu && (
          <>
            <Box flexGrow={1} />
            {props.menu}
          </>
        )}
      </Toolbar>
    </AppBar>
  );
};

export type { Props as ElHeaderProps };
