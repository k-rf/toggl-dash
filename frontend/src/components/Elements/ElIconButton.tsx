import { IconButton, IconButtonProps } from "@mui/material";

type Props = Pick<
  IconButtonProps,
  "children" | "onClick" | "onMouseDown" | "onMouseUp" | "onMouseLeave" | "disableTouchRipple"
>;

export const ElIconButton = (props: Props) => {
  return <IconButton {...props} />;
};

export type { Props as ElIconButtonProps };
