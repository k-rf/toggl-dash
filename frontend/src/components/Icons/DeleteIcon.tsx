import { DeleteForever } from "@mui/icons-material";
import { SvgIconProps } from "@mui/material";

type Props = Pick<SvgIconProps, "color" | "fontSize">;

export const DeleteIcon = (props: Props) => {
  return <DeleteForever {...props} />;
};
