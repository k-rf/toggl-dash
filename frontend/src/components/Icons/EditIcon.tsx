import { Edit } from "@mui/icons-material";
import { SvgIconProps } from "@mui/material";

type Props = Pick<SvgIconProps, "color" | "fontSize">;

export const EditIcon = (props: Props) => {
  return <Edit {...props} />;
};
