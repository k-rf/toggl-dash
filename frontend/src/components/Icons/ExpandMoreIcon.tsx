import { ExpandMore } from "@mui/icons-material";
import { SvgIconProps } from "@mui/material";

type Props = Pick<SvgIconProps, "color" | "fontSize">;

export const ExpandMoreIcon = (props: Props) => {
  return <ExpandMore {...props} />;
};
