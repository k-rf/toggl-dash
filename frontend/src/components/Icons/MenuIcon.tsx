import { Menu } from "@mui/icons-material";

export const MenuIcon = () => {
  return <Menu />;
};
