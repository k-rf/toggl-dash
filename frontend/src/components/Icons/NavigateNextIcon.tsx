import { NavigateNext } from "@mui/icons-material";
import { SvgIconProps } from "@mui/material";

type Props = Pick<SvgIconProps, "color" | "fontSize">;

export const NavigateNextIcon = (props: Props) => {
  return <NavigateNext {...props} />;
};
