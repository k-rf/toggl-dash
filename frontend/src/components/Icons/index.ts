export * from "./ExpandMoreIcon";
export * from "./PlayArrowIcon";
export * from "./VisibilityIcon";
export * from "./VisibilityOffIcon";
