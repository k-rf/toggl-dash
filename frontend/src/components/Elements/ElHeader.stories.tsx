import type { Meta, StoryObj } from "@storybook/react";

import { MenuIcon } from "../Icons";

import { ElHeader, ElHeaderProps } from "./ElHeader";
import { ElIconButton } from "./ElIconButton";

export default { component: ElHeader, args: { title: "Header" } } as Meta<ElHeaderProps>;

type Story = StoryObj<ElHeaderProps>;

export const Default: Story = {};

export const WithMenu: Story = {
  args: {
    menu: (
      <ElIconButton color="inherit">
        <MenuIcon />
      </ElIconButton>
    ),
  },
};
