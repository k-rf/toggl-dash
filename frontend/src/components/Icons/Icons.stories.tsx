import { Box, Stack } from "@mui/material";
import type { Meta, StoryObj } from "@storybook/react";

import { ElPaper } from "../Elements";
import { ElTypography } from "../Elements/ElTypography";

import { ExpandMoreIcon } from "./ExpandMoreIcon";
import { PlayArrowIcon } from "./PlayArrowIcon";
import { VisibilityIcon } from "./VisibilityIcon";
import { VisibilityOffIcon } from "./VisibilityOffIcon";

export default {
  title: "Components/Icons",
  render: () => {
    return (
      <Box display="flex" flexWrap="wrap">
        <SxPaper title="Visibility">
          <VisibilityIcon />
        </SxPaper>
        <SxPaper title="VisibilityOff">
          <VisibilityOffIcon />
        </SxPaper>
        <SxPaper title="PlayArrow">
          <PlayArrowIcon />
        </SxPaper>
        <SxPaper title="ExpandMore">
          <ExpandMoreIcon />
        </SxPaper>
        {/* NOTE: プロダクトで利用するアイコンが増えたらここに追記する */}
      </Box>
    );
  },
} as Meta;

export const Default: StoryObj = {};

const SxPaper = ({ title, children }: { title: string; children: React.ReactNode }) => {
  return (
    <ElPaper
      sx={{
        display: "flex",
        width: 96,
        height: 48,
        justifyContent: "center",
        alignItems: "center",
        m: (theme) => theme.spacing(0, 2, 2, 0),
      }}
    >
      <Stack>
        <Box display="flex" justifyContent="center">
          {children}
        </Box>
        <ElTypography variant="caption">{title}</ElTypography>
      </Stack>
    </ElPaper>
  );
};
