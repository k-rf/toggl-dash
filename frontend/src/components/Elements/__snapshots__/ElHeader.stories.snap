// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Elements/ElHeader > Default 1`] = `
<body
  style=""
>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-vv2e0-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        Header
      </div>
    </header>
  </div>
</body>
`;

exports[`Components/Elements/ElHeader > WithMenu 1`] = `
<body
  style=""
>
  <div>
    <header
      class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-vv2e0-MuiPaper-root-MuiAppBar-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-hyum1k-MuiToolbar-root"
      >
        Header
        <div
          class="MuiBox-root css-i9gxme"
        />
        <button
          class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
          tabindex="0"
          type="button"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </header>
  </div>
</body>
`;
